packages/element/src/serialize.js

```diff
@@ -494,10 +501,22 @@ export function renderComponent(
 	context,
 	legacyContext = {}
 ) {
-	const instance = new Component( props, legacyContext );
+	const instance = new /** @type {import('react').ComponentClass} */ ( Component )(
+		props,
+		legacyContext
+	);
 
-	if ( typeof instance.getChildContext === 'function' ) {
-		Object.assign( legacyContext, instance.getChildContext() );
+	if (
+		typeof (
+			// Ignore reason: Current prettier reformats parens and mangles type assertion
+			// prettier-ignore
+			/** @type {{getChildContext?: () => unknown}} */ ( instance ).getChildContext
```

Comment:
Can you help me understand the difference between `=> unknown` and `=> *` (`any`)?

Reply:
There's some documentation on [`unknown` here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type):

> TypeScript 3.0 introduces a new top type `unknown`. `unknown` is the type-safe counterpart of `any`. Anything is assignable to `unknown`, but `unknown` isn’t assignable to anything but itself and `any` without a type assertion or a control flow based narrowing. Likewise, no operations are permitted on an `unknown` without first asserting or narrowing to a more specific type.

The big difference is that `any` can be used as _anything_, whereas `unknown` types require more defensive usage. An example may be best:

```js
/** @type {any} */
const x = 'a string';

// Any satisfies _anything_, let's pretend it's a promise:
x.then( console.log ); // ✅Cool beans. Have fun at runtime 😎 

/** @type {unknown} */
const y = 'a string';

const badLen = y.length; // 💥Error, it's unknown! We're forced to be much more careful…
if ( typeof y === 'string' ) {
  const goodLen = y.length; // Oh, it's a string, go ahead 👍 
}
```